pkg/storage/stores/shipper/compactor/deletion/delete_request.go

```diff
@@ -52,19 +93,30 @@ func (d *DeleteRequest) IsDeleted(entry retention.ChunkEntry) (bool, []model.Int
 		return true, nil
```

Comment:
I think we should be returning a non `nil` when the whole chunk is covered by a delete request which has a line filter? Without a filter, it would delete the whole chunk.
Let us also add a test case for that as well?

Reply:
I think you can use [HasFilter](https://github.com/grafana/loki/blob/79f0e349f56c3ec6663a3dc0457173c49a2765b4/pkg/logql/syntax/ast.go#L41) to decide whether `FilterFunc` is required or not. I think we should cover the cases of both a chunk being fully or partially deleted with/without a filter.

Reply:
Good catch. The check using `HasFilter` has been added plus an extra test for this. 

I've also added 1 test for a partial deletion with a filter. The code didn't have to change so this 1 test will suffice hopefully.